Bottom: 0110b8e12cf2db054de21bb45dda37db98997557
Top:    bc98e32c7ce9614150acb9f1e6dbefa1762c1634
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-05-22 20:44:20 -0400

Refresh of mpc5200-ac97

---

diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 39af12d..7253303 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -362,15 +362,19 @@ reset:
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) != 0) {
 		udelay(1);
-		if (timeout++ > 1000)
+		if (timeout++ > 1000) {
+			psc_ac97_warm_reset(&ac97);
 			goto reset;
+		}
 	}
 	/* then wait for the transition to high */
 	timeout = 0;
 	while ((in_8(&regs->ipcr_acr.ipcr) & 0x80) == 0) {
 		udelay(1);
-		if (timeout++ > 1000)
+		if (timeout++ > 1000) {
+			psc_ac97_warm_reset(&ac97);
 			goto reset;
+		}
 	}
 
 	out_8(&regs->command, MPC52xx_PSC_TX_ENABLE | MPC52xx_PSC_RX_ENABLE);
